### ISSUE TITLE
[BUG] DiskWriterQueue is not reliable when handling back-to-back insertions.

### ISSUE THREAD

**Issue by kaan-kaya** [mbdavid/LiteDB#2127]
**Version**
LiteDB 5.0.11 - .NET Core 3.1

**Describe the bug**
In back to back insertions, if the second insertion comes in when the thread processing the first insertion has exited the dequeue loop but the task is not complete yet, the second item is not persisted to disk until a new item comes in or the LiteDatabase instance is disposed. This would mean that the second item's data could be lost if there's a server/process crash.

**Code to Reproduce**
Sample class to be stored:
```
    public class ExampleItem
    {
        [BsonId]
        public Guid Id { get; set; }

        public string SomeProperty { get; set; }
    }

```
A class that holds the LiteDatabase instance and does insertion:
```
    public sealed class ExampleItemRepository : IDisposable
    {
        public const string DatabaseFileName = "SampleDb";

        private readonly LiteDatabase _liteDb;

        public ExampleItemRepository(string databasePath)
        {
            var connectionString = new ConnectionString
            {
                Filename = databasePath,
                Connection = ConnectionType.Direct
            };

            _liteDb = new LiteDatabase(connectionString);
        }

        public void Insert(ExampleItem item)
        {
            var collection = _liteDb.GetCollection<ExampleItem>();
            collection.Insert(item);
        }

        public void Dispose()
        {
            _liteDb?.Dispose();
        }
    }
```
and finally a unit test (using NUnit 3.1.12) to reproduce:
```
    [TestFixture]
    public class ReproTests
    {
        [Test]
        public void InsertItemBackToBack_Test()
        {
            var databaseDirectory = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "DatabaseLocation");
            var databasePath = Path.Combine(databaseDirectory, "SampleDatabase.db");
            var databaseLogPath = Path.Combine(databaseDirectory, "SampleDatabase-log.db");
            
            // Repeat in a loop to hit that scenario where queue processing thread exists the loop but the task still remains incomplete.
            for (var i = 0; i < 100; i++)
            {
                if (Directory.Exists(databaseDirectory))
                {
                    Directory.Delete(databaseDirectory, true);
                }
                Directory.CreateDirectory(databaseDirectory);

                using var subject = new ExampleItemRepository(databasePath);

                var item1 = new ExampleItem
                {
                    Id = Guid.NewGuid(),
                    SomeProperty = Guid.NewGuid().ToString()
                };
                var item2 = new ExampleItem
                {
                    Id = Guid.NewGuid(),
                    SomeProperty = Guid.NewGuid().ToString()
                };
                
                subject.Insert(item1);
                subject.Insert(item2);

                // Allow items to be processed.
                Thread.Sleep(1000);

                var liteDbPath = databasePath;
                var liteDbLogPath = databaseLogPath;
                var copiedLiteDbPath = liteDbPath + ".copy";
                var copiedLiteDbLogPath = liteDbLogPath + ".copy";
                File.Copy(liteDbPath, copiedLiteDbPath);
                File.Copy(liteDbLogPath, copiedLiteDbLogPath);
                var liteDbContent = File.ReadAllText(copiedLiteDbPath, Encoding.UTF8);
                liteDbContent += File.ReadAllText(copiedLiteDbLogPath, Encoding.UTF8);

                Assert.True(liteDbContent.Contains(item1.SomeProperty, StringComparison.OrdinalIgnoreCase), $"Could not find item 1 property. {item1.SomeProperty}, Iteration: {i}");
                Assert.True(liteDbContent.Contains(item2.SomeProperty, StringComparison.OrdinalIgnoreCase), $"Could not find item 2 property. {item2.SomeProperty}, Iteration: {i}");
            }
        }
    }
```

**Expected behavior**
The queue could be more reliable to process the second item.

**Proposal for fix**
Please review this pull request which contains a more reliable DiskWriterQueue to process the items:
https://github.com/mbdavid/LiteDB/pull/2126

**Comment by kcsombrio** [comment]
Pull request accepted, thanks for your support

**Comment by kaan-kaya** [comment]
@kcsombrio hi, is there an ETA on a new release where this would be included?

**Comment by kcsombrio** [comment]
Hi @kaan-kaya, sorry for the delay, we just released v5.0.12 with this fix included, thanks for the support. Rgds
